resource/resourceadapters/charmstore.go

```diff
@@ -110,7 +119,8 @@ func (client CSRetryClient) GetResource(req charmstore.ResourceRequest) (charmst
 	var lastErr error
 	args.NotifyFunc = func(err error, i int) {
 		// Remember the error we're hiding and then retry!
-		logger.Debugf("(attempt %d) retrying resource download from charm store due to error: %v", i, err)
+		logger.Warningf("(attempt %d) retrying resource %q download from charm store [channel (%v), charm (%v), resource revision (%v)] due to error: %v",
```

Comment:
does the code already handle reporting "attempt 0" to the user?
In the past we've found it useful to give attempt X/Y so they know we'll stop trying eventually, but if that information is hard to find, then we can skip it.